presto-native-execution/presto_cpp/main/PeriodicTaskManager.cpp

```diff
@@ -129,16 +126,16 @@ void PeriodicTaskManager::start() {
         "clean_old_tasks");
   }
 
-  if (auto* mappedMemory = velox::memory::MappedMemory::getInstance()) {
+  if (auto* allocator = velox::memory::MemoryAllocator::getInstance()) {
     scheduler_.addFunction(
-        [mappedMemory]() {
+        [allocator]() {
           REPORT_ADD_STAT_VALUE(
-              kCounterMappedMemoryBytes, (mappedMemory->numMapped() * 4096l));
+              kCounterMappedMemoryBytes, (allocator->numMapped() * 4096l));
```

Comment:
You will be calling a pure virtual function if not mmap impled?

Reply:
The non-mapped one has an implementation for this.

Reply:
Ah okay we still have this MemoryAllocatorImpl class. I thought they were merged to just MemoryAllocator alone.